src/test/java/com/configcat/DataGovernanceTest.java

```diff
@@ -0,0 +1,224 @@
+        secondServer.shutdown();
+    }
+
+    @Test
+    public void shouldRedirectToAnotherServerWhenForced() throws IOException, ExecutionException, InterruptedException {
+        MockWebServer firstServer = this.createServer();
+        MockWebServer secondServer = this.createServer();
+        ConfigFetcher fetcher = this.createFetcher(firstServer.url("/").toString(), false);
+
+        // Arrange
+        String firstBody = String.format(JsonTemplate, secondServer.url("/"), 2);
+        firstServer.enqueue(new MockResponse().setResponseCode(200).setBody(firstBody));
+
+        String secondBody = String.format(JsonTemplate, secondServer.url("/"), 0);
+        secondServer.enqueue(new MockResponse().setResponseCode(200).setBody(secondBody));
+
+        // Act
+        FetchResponse response = fetcher.getConfigurationJsonStringAsync().get();
+
+        // Assert
+        assertEquals(secondBody, response.config());
+        assertEquals(1, firstServer.getRequestCount());
+        assertEquals(1, secondServer.getRequestCount());
+
+        // Cleanup
+        fetcher.close();
+        firstServer.shutdown();
+        secondServer.shutdown();
+    }
+
+    @Test
+    public void shouldBreakTheRedirectLoop() throws IOException, ExecutionException, InterruptedException {
+        MockWebServer firstServer = this.createServer();
+        MockWebServer secondServer = this.createServer();
+        ConfigFetcher fetcher = this.createFetcher(firstServer.url("/").toString(), false);
+
+        // Arrange
+        String firstBody = String.format(JsonTemplate, secondServer.url("/"), 1);
+        firstServer.enqueue(new MockResponse().setResponseCode(200).setBody(firstBody));
+        firstServer.enqueue(new MockResponse().setResponseCode(200).setBody(firstBody));
+
+        String secondBody = String.format(JsonTemplate, firstServer.url("/"), 1);
+        secondServer.enqueue(new MockResponse().setResponseCode(200).setBody(secondBody));
+
+        // Act
+        FetchResponse response = fetcher.getConfigurationJsonStringAsync().get();
+
+        // Assert
+        assertEquals(firstBody, response.config());
+        assertEquals(2, firstServer.getRequestCount());
+        assertEquals(1, secondServer.getRequestCount());
+
+        // Cleanup
+        fetcher.close();
+        firstServer.shutdown();
+        secondServer.shutdown();
+    }
+
+    @Test
+    public void shouldRespectCustomUrlWhenNotForced() throws IOException, ExecutionException, InterruptedException {
+        MockWebServer firstServer = this.createServer();
+        MockWebServer secondServer = this.createServer();
+        ConfigFetcher fetcher = this.createFetcher(firstServer.url("/").toString(), true);
+
+        // Arrange
+        String firstBody = String.format(JsonTemplate, secondServer.url("/"), 1);
+        firstServer.enqueue(new MockResponse().setResponseCode(200).setBody(firstBody));
+
+        String secondBody = String.format(JsonTemplate, secondServer.url("/"), 0);
```

Comment:
i think this is not needed

Reply:
already removed [here](https://github.com/configcat/java-sdk/pull/3/commits/107ec73c62175f427b6317655e8ca76b76cf10a4)